design/dsl/endpoint_test.go

```diff
@@ -151,7 +151,7 @@ func assertEndpointMetaData(t *testing.T, actual design.MetadataExpr, expected d
 
 func assertEndpointRequestResponse(t *testing.T, assertType string, actual design.DataType, expected *design.UserTypeExpr) {
 	ut, ok := actual.(*design.UserTypeExpr)
-	if !ok {
+	if !ok || ut == nil {
```

Comment:
perhaps adding a return would be enough, as it would seem there is little point to continuing with the other assertions in the function at that point?